components/bl-popover-component/src/helpers.js

```diff
@@ -1,45 +1,72 @@
+  const targetVerticalCenter = targetRef.clientWidth / 2;
+  const contentVerticalCenter = contentElement.clientWidth / 2;
 
-  const horizontalTopShift = contentOffset.top - rootOffset.top + contentHorizontalCenter - rootHorizontalCenter;
-  const verticalLeftShift = contentOffset.left + contentContainerVerticalCenter - rootVerticalCenter;
+  const horizontalTopShift = targetOffset.top - contentOffset.top + targetHorizontalCenter - contentHorizontalCenter;
+  const verticalLeftShift = targetOffset.left + targetVerticalCenter - contentVerticalCenter;
 
+  const {
+    width: targetWidth, height: targetHeight,
+    x: targetX, y: targetY,
+    right: targetRight, bottom: targetBottom
+  } = targetRef.getBoundingClientRect();
   const { width: contentWidth, height: contentHeight } = contentElement.getBoundingClientRect();
-  const { width: rootWidth, height: rootHeight } = root.getBoundingClientRect();
 
   const ShiftHandler = {
     [Position.RIGHT] : {
       topShift : horizontalTopShift,
-      leftShift: contentOffset.left + contentWidth + root.firstChild.clientHeight + rootOffset.left
+      leftShift: targetOffset.left + targetWidth + contentElement.firstChild.clientHeight + contentOffset.left
     },
     [Position.LEFT]  : {
       topShift : horizontalTopShift,
-      leftShift: contentOffset.left - rootWidth - root.firstChild.clientHeight + rootOffset.left
+      leftShift: targetOffset.left - contentWidth - contentElement.firstChild.clientHeight + contentOffset.left
     },
     [Position.TOP]   : {
-      topShift : contentOffset.top - rootOffset.top - rootHeight - root.firstChild.clientHeight,
+      topShift : targetOffset.top - contentOffset.top - contentHeight - contentElement.firstChild.clientHeight,
       leftShift: verticalLeftShift
     },
     [Position.BOTTOM]: {
-      topShift : contentOffset.top + contentHeight - rootOffset.top + root.firstChild.clientHeight,
+      topShift : targetOffset.top + targetHeight - contentOffset.top + contentElement.firstChild.clientHeight,
       leftShift: verticalLeftShift
     }
   };
 
-  return ShiftHandler[position];
+  const PositionCheck = {
+    left  : () => (
+      targetX >= contentWidth
+        && (targetY + targetHorizontalCenter) >= contentHorizontalCenter
+        && window.innerHeight - targetBottom + targetHorizontalCenter >= contentHorizontalCenter
+    ),
+    right : () => (
+      document.documentElement.clientWidth - targetRight >= contentWidth
+        && (targetY + targetHorizontalCenter) >= contentHorizontalCenter
+        && window.innerHeight - targetBottom + targetHorizontalCenter >= contentHorizontalCenter
+    ),
+    top   : () => (
+      targetY >= contentHeight
+        && (targetX + targetVerticalCenter) >= contentVerticalCenter
+        && document.documentElement.clientWidth - targetRight + targetVerticalCenter >= contentVerticalCenter
+    ),
+    bottom: () => (
+      window.innerHeight - targetBottom >= contentHeight
```

Comment:
I'm not sure if it good approach, and my offer is to do it(check for intersection) through IntersectionObserver, but I'm not insisting.